### PR TITLE
Add `GetAllItems` with an extra parameter.

### PR DESCRIPTION
**Pull request by ricaun**
I create an extra `GetAllItems`  method to work with `pageGetter` function when an extra parameter. Like `GetAppBundleVersionsAsync(string id, Page page = null)`.
```C#
await DesignAutomationClient.GetAllItems(DesignAutomationClient.GetAppBundleVersionsAsync, id);
```
Added documentation as well using ChatGPT to generate the `<summary>`.

### CODE REVIEW

File: src/Autodesk.Forge.DesignAutomation/ApiClient.cs [hunk 1]
```diff
@@ -143,6 +149,28 @@ public async Task<List<T>> GetAllItems<T>(Func<string, Task<Page<T>>> pageGetter
...
+            string paginationToken = null;
+            do
+            {
+                var resp = await pageGetter(parameter, paginationToken);
+                paginationToken = resp.PaginationToken;
+                ret.AddRange(resp.Data);
+            }
+            while (paginationToken != null);
+            return ret;
+        }
```
Comment (zhuliice) [inline src/Autodesk.Forge.DesignAutomation/ApiClient.cs:173]: Could you also add unit tests for this new function? 

See the example: https://github.com/Autodesk-Forge/forge-api-dotnet-design.automation/blob/2672f04bf625c106ee5c5e742e710715db74bf30/tests/e2e/TestAppBundlesApi.cs#L61-L71

Comment (ricaun) [reply src/Autodesk.Forge.DesignAutomation/ApiClient.cs:173]: Yes, but to test locally? Is using the default environment variable?

**Comment by szilvaa** [comment]
@ricaun  Passing this extra parameter appears to be unnecessary because the callback can store this as a state variable. The example given in above could be written like this:
```C#
await daClient.GetAllItems(async page => await daClient.GetAppBundleVersionsAsync(id, page));
```

**Comment by ricaun** [comment]
You are right, could be done like that. 
Regarding that the tests of the methods GetAppBundleAliasesAsync and GetAppBundleVersionsAsync fail...

**Comment by zhuliice** [comment]
Thanks for the effort, @ricaun! 
I think we can just close this PR then?

**Comment by szilvaa** [comment]
>Regarding that the tests of the methods GetAppBundleAliasesAsync and GetAppBundleVersionsAsync fail

They fail because you provided a "failed" response in the JSON file (tests/e2e/recordings/AppBundles_GetAllAlias.json). These JSON files are generated when you first create a test (or you can manually author them). If you rely on recording them then you must be provide valid  credentials (in an appsettings.user.json file).

**Comment by ricaun** [comment]
> > Regarding that the tests of the methods GetAppBundleAliasesAsync and GetAppBundleVersionsAsync fail
> 
> They fail because you provided a "failed" response in the JSON file (tests/e2e/recordings/AppBundles_GetAllAlias.json). These JSON files are generated when you first create a test (or you can manually author them). If you rely on recording them then you must be provide valid credentials (in an appsettings.user.json file).

I believe this test does not use credentials, `Autodesk.Forge.Core.E2eTestHelpers` creates a fake forge API or something.

I'm using `GetAppBundleAliasesAsync` and `GetAppBundleVersionsAsync` with some valid credentials and work fine, just reporting that the test fails.

And gonna fail if you use `await daClient.GetAllItems(async page => await daClient.GetAppBundleVersionsAsync(id, page));` as well.

> Thanks for the effort, @ricaun! I think we can just close this PR then?

Yes, I gonna close the PR.

**Pull request closed without merging**